app/src/main/java/ch/epfl/polybazaar/UI/FillListing.java

```diff
@@ -128,44 +120,46 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
         if(resultCode != Activity.RESULT_OK){
             pictureView.setTag(-1);
         }
-        else if (requestCode == RESULT_LOAD_IMAGE){
+        else if (requestCode == LOAD_IMAGE){
             if(data == null){
                 pictureView.setTag(-1);
                 return;
             }
-            Uri selectedImage = data.getData();
-            Bitmap bitmap;
-            try {
-                bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), selectedImage);
-            } catch (IOException e) {
-                e.printStackTrace();
-                Toast.makeText(this, R.string.unable_load_image, Toast.LENGTH_SHORT).show();
-                return;
-            }
-            stringImage = convertBitmapToStringWithQuality(bitmap, QUALITY);
-            imageManager.addImage(listStringImage, stringImage);
-            imageManager.updateViewPagerVisibility(listStringImage);
-        }
-        else if (requestCode == RESULT_TAKE_PICTURE){
-           stringImage = convertFileToStringWithQuality(imageManager.getPhotoFile(), QUALITY);
-           imageManager.addImage(listStringImage, stringImage);
-           imageManager.updateViewPagerVisibility(listStringImage);
+            getNewImage(data);
         }
-        else if (requestCode == RESULT_ADD_MP) {
+        else if (requestCode == TAKE_IMAGE){
             if (data != null) {
-                if (data.getBooleanExtra(VALID, false)) {
-                    lng = data.getDoubleExtra(LNG, NOLNG);
-                    lat = data.getDoubleExtra(LAT, NOLAT);
-                    addMP.setText(R.string.change_MP);
-                } else {
-                    lng = NOLNG;
-                    lat = NOLAT;
-                    addMP.setText(R.string.add_MP);
-                }
+                getNewImage(data);
+            }
+        }
+        if (resultCode == Activity.RESULT_OK && ADD_MP == requestCode) {
```

Comment:
To keep the structure, this could be an else if

Reply:
I had that at first, but for some reason, it would never enter the else if, but it enters the if..